### PR TITLE
Synchronize main concord code base to March release

### PR DESCRIPTION
**Pull request by no2chem**
This PR synchronizes the current code base to the March release of Concord.

### CODE REVIEW

File: communication/src/main/proto/thin_replica.proto [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+//
+// Note: Updates are not blocks but KV-pairs that belong to the blockchain.
+//
+
+syntax = "proto3";
+package com.vmware.concord.thin_replica;
+
+import "google/protobuf/empty.proto";
+
+service ThinReplica {
```
Comment (MRHarrison) [inline communication/src/main/proto/thin_replica.proto:15]: Will the build fail if we remove this?

File: communication/src/main/proto/concord.proto [hunk 2]
```diff
@@ -186,6 +264,19 @@ message ProtocolResponse {
    optional int64 net_version = 2;
 }
 
+message PruneRequest {
```
Comment (MRHarrison) [inline communication/src/main/proto/concord.proto:267]: Same here will the build fail if we remove these?

File: docker/compose/simple4.yml [hunk 3]
```diff
@@ -1,10 +1,5 @@
 version: '3'
 services:
-  concord-truffle:
```
Comment (MRHarrison) [inline docker/compose/simple4.yml:3]: Is this needed for the tutorial and connecting a console?

**Comment by no2chem** [comment]
@MRHarrison addressed all review comments

**Comment by MRHarrison** [comment]
@no2chem  Since you found the issue to be truffle, I think we should go ahead merge this in.

**Pull request merged**